src/main/java/com/arpnetworking/metrics/mad/parsers/GraphitePlaintextToRecordParser.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright 2017 Inscope Metrics, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.mad.parsers;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.metrics.common.parsers.Parser;
+import com.arpnetworking.metrics.common.parsers.exceptions.ParsingException;
+import com.arpnetworking.metrics.mad.model.DefaultMetric;
+import com.arpnetworking.metrics.mad.model.DefaultRecord;
+import com.arpnetworking.metrics.mad.model.Record;
+import com.arpnetworking.tsdcore.model.MetricType;
+import com.arpnetworking.tsdcore.model.Quantity;
+import com.google.common.base.Charsets;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import net.sf.oval.constraint.NotNull;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.nio.ByteBuffer;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Parses Graphite plaintext data as a {@link Record}.
+ *
+ * https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol
+ *
+ * In addition to the specified features, this parser supports multiple
+ * datagrams separated by newlines.
+ *
+ * The source also allows you to add dimensions in three ways:
+ *
+ * 1) Specifying global tags programmatically.
+ * 2) Parsing tags from the metric name in Carbon format.
+ *     See: http://graphite.readthedocs.io/en/latest/tags.html
+ * 3) Parsing tags from the metric name in InfluxDb format.
```

Comment:
Maybe I missed something, but I didn't see this in the code.

Reply:
As per MR comments this is not yet implemented. I will update the comment.